Sources/Utility/ArgumentParser.swift

```diff
@@ -356,6 +400,23 @@ public final class ArgumentParser: ArgumentParserProtocol {
                 guard let positionalArg = positionalArgsIterator.next() else {
                     throw ArgumentParserError.unexpectedArgument(arg)
                 }
+
+                // If this is a subparser argument.
+                if positionalArg.subparserArg {
+                    guard let subparser = self.subparser else {
+                        fatalError("Unexpected subparser argument when the subparser doesn't exists.")
+                    }
+                    guard let theParser = subparser.parsers[arg] else {
+                        throw ArgumentParserError.expectedArguments(Array(subparser.parsers.keys))
+                    }
+                    // Set subparser result as the argument which was chosen.
+                    subparser.result = arg
+                    // Parse remaining arguments using the subparser.
+                    let subparserResult = try theParser.parse(Array(argumentsIterator))
+                    // Merge the results.
+                    result.merge(subparserResult)
```

Comment:
I wonder if it makes more sense to chain the results, rather than merge them. We can still make the query methods search through the chain.

This would let the result remember explicitly which parser it came from.